thunder/core/utils.py

```diff
@@ -1128,11 +1128,21 @@ def find_producer_symbols(trace: TraceCtx, proxies: Sequence[Proxy], stop_proxie
         if p is not None:
             result.add(p)
             for arg in p.flat_args:
-                arg_name = arg.name if isinstance(arg, Proxy) else None
+                if not isinstance(arg, Proxy):
+                    continue
+                arg_name = arg.name
                 if arg_name not in map(lambda x: x.name, stop_proxies) and arg_name not in seen:
                     queue.append(arg)
                     seen.add(arg_name)
-    original_order = {bsym: i for i, bsym in enumerate(trace.bound_symbols)}
+    original_order = dict()
+    for i, bsym in enumerate(trace.bound_symbols):
+        # Don't overwrite the order if it's already encountered. This is necessary for
```

Comment:
This seems like a good fix and comment, but the comment assumes the reader knows the previous code. I think we could add a little more context here, like "Creates a map from ..."

Like, what's the point of ordering, or why might there be duplicate bsyms? 

Also, would the description of what this function returns benefit from being updated to explain that the bsyms in the tuple are presented in order, and are the "first" bsyms that produce a proxy if there are multiple such symbols?